### PR TITLE
decision issue nonrating

### PR DESCRIPTION
**Pull request by jiahuang**
Connects #7978

* Updates nonrating decision issues to have the correct date & description
* removes decision issues associated with appeals from showing up 

screen shot 2018-12-12 at 11 18 17 am

### CODE REVIEW

File: app/models/decision_issue.rb [hunk 1]
```diff
@@ -22,14 +22,28 @@ def approx_decision_date
     profile_date ? profile_date.to_date : end_product_last_action_date
   end
 
-  def issue_category
-    # TODO: figure out how to calculate this
-    return unless request_issues.any?
-    request_issues.first.issue_category
+  def serialized_description
```
Comment (shanear) [inline app/models/decision_issue.rb]: I don't think this is the meaning of `serialized`. Just calling this `description` would work.

File: app/models/decision_issue.rb [hunk 2]
```diff
@@ -22,14 +22,28 @@ def approx_decision_date
...
+  end
+
+  def nonrating_description
+    return unless associated_request_issue&.nonrating?
+    "#{disposition}: #{associated_request_issue.issue_category} - #{associated_request_issue.description}"
+  end
+
+  def rating_description
+    return description if description
+    return "#{decision_text}" unless associated_request_issue&.notes
```
Comment (shanear) [inline app/models/decision_issue.rb]: no need to wrap this in quotes and `#{}`

File: app/models/decision_issue.rb [hunk 3]
```diff
@@ -22,14 +22,28 @@ def approx_decision_date
...
+    request_issues.first
+  end
+
+  def nonrating_description
+    return unless associated_request_issue&.nonrating?
+    "#{disposition}: #{associated_request_issue.issue_category} - #{associated_request_issue.description}"
+  end
+
+  def rating_description
+    return description if description
```
Comment (shanear) [inline app/models/decision_issue.rb]: shouldn't we do this for both rating and nonrating?

File: app/models/decision_issue.rb [hunk 4]
```diff
@@ -22,14 +22,28 @@ def approx_decision_date
...
 
   private
 
+  def associated_request_issue
+    return unless request_issues.any?
+    request_issues.first
+  end
+
+  def nonrating_description
+    return unless associated_request_issue&.nonrating?
```
Comment (shanear) [inline app/models/decision_issue.rb]: i think this reads better if you make the logic happen in a terinary operator in description:

```
def description
  associated_request_issue&.nonrating? nonrating_issue_description : rating_issue_description
end
```

File: spec/support/intake_helpers.rb [hunk 5]
```diff
@@ -171,20 +171,52 @@ def setup_legacy_opt_in_appeals(veteran_file_number)
...
-                                      participant_id: veteran.participant_id,
-                                      decision_text: "contested supplemental claim decision issue",
-                                      profile_date: Time.zone.now - 2.days,
-                                      promulgation_date: Time.zone.now - 2.days,
-                                      benefit_type: supplemental_claim_with_decision_issues.benefit_type)
+    nonrating_request_issue = setup_request_issue_with_nonrating_decision_issue(supplemental_claim_with_decision_issues)
+    rating_request_issue = setup_request_issue_with_rating_decision_issue(supplemental_claim_with_decision_issues)
 
-    contested_decision_issue
+    rating_request_issue.decision_issues + nonrating_request_issue.decision_issues
```
Comment (shanear) [inline spec/support/intake_helpers.rb]: Nice tests 🎆

File: spec/factories/request_issue.rb [hunk 6]
```diff
@@ -8,6 +8,40 @@
       end_product_establishment { create(:end_product_establishment) }
     end
 
+    trait :with_rating_decision_issue do
+      transient do
+        veteran_participant_id ""
```
Comment (pkarman) [inline spec/factories/request_issue.rb]: why setting this to empty string?

File: spec/factories/request_issue.rb [hunk 7]
```diff
@@ -8,6 +8,40 @@
...
+          profile_date: request_issue.rating_issue_profile_date,
+          benefit_type: request_issue.review_request.benefit_type,
+          decision_text: "rating decision issue",
+          request_issues: [request_issue])
+      end
+    end
+
+    trait :with_nonrating_decision_issue do
+      transient do
+        veteran_participant_id ""
```
Comment (pkarman) [inline spec/factories/request_issue.rb]: same comment about empty string

**Pull request merged**